app/models/supermarket.rb

```diff
@@ -1,3 +1,7 @@
 class Supermarket < ApplicationRecord
   has_many :items
+
+  def distinct_customers
+    Customer.joins(:items).where(items: {supermarket_id: self.id}).distinct
```

Comment:
Excellent join!! Great work remembering `distinct` at the end to avoid duplication.

Another option:
- add a new association to this model to associate directly to customers, since you can build off the items association that is already present: `has_many :customers, through: :items`
- with this new association, you can do `self.customers.distinct` in a model method

Your implementation and the idea above are equally great solutions.